### PR TITLE
Add Java signature verification.

### PR DESCRIPTION
**Pull request by thmsbinder**
- Add implementation of signature verification
- Add more tests
- Add automatic raw/PEM format detection for command line tools
- Fix essential typo: logID not logId - variable names are taken literally as JSON keys
- Fix capitalization in javadocs per style guide

Ref #2854

### CODE REVIEW

File: java/src/main/java/com/google/oak/transparency/LogEntryVerifier.java [hunk 1]
```diff
@@ -40,6 +41,12 @@ public static void main(String[] args) throws Exception {
     byte[] publicKeyBytes = Files.readAllBytes(Path.of(args[1]));
     byte[] endorsementBytes = Files.readAllBytes(Path.of(args[2]));
 
+    // Auto-detect PEM format of public key.
+    String publicKeyString = new String(publicKeyBytes, StandardCharsets.UTF_8);
+    if (publicKeyString.startsWith("-----BEGIN PUBLIC KEY-----")) {
```
Comment (tiziano88) [inline java/src/main/java/com/google/oak/transparency/LogEntryVerifier.java:46]: Is this constant not defined anywhere that we already depend on?

File: java/src/main/java/com/google/oak/transparency/SignatureVerifier.java [hunk 2]
```diff
@@ -49,24 +61,51 @@ private static Optional<Failure> failure(String message) {
...
+        }
         if (contentBytes == null || contentBytes.length == 0) {
             return failure("empty content");
         }
-        if (publicKeyBytes == null || publicKeyBytes.length == 0) {
-            return failure("empty public key");
+
+        X509EncodedKeySpec keySpec = new X509EncodedKeySpec(publicKeyBytes);
+        Exception exception = null;
+        boolean outcome = false;
```
Comment (tiziano88) [inline java/src/main/java/com/google/oak/transparency/SignatureVerifier.java:83]: Given that the return types are already not always semantically aligned (e.g. sometimes option.empty means success, sometimes failure), I suggest using a clearer name for this variable, e.g. `success`

**Review by tiziano88 — APPROVED** [review]

File: java/src/main/java/com/google/oak/transparency/SignatureVerifier.java (hunk 2 above)
Comment (thmsbinder) [reply java/src/main/java/com/google/oak/transparency/SignatureVerifier.java:83]: Right, done.

File: java/src/main/java/com/google/oak/transparency/LogEntryVerifier.java (hunk 1 above)
Comment (thmsbinder) [reply java/src/main/java/com/google/oak/transparency/LogEntryVerifier.java:46]: Couldn't find any lib for this. But at least avoiding repetition in many places.

**Pull request merged**